controllers/user.controller.js

```
@@ -1,47 +1,49 @@
 const { where, json } = require('sequelize');
-const { User, Post } = require('../models/index');
+const { User, Post, Role, Permission } = require('../models/index');
 const bcrypt = require('bcryptjs');
 const jwt = require('jsonwebtoken');
 
-const SECRET_KEY = 'your_secret_key';
-
 module.exports = {
-    createUser: async (req, res, next) => {
-        // try {
-        //     const { name ,username,password } = req.body;
-        //     const myObject = await User.create({
-        //         name,username,password
-        //     });
-
-        //     res.status(200).json(myObject);
-        // } catch (error) {
-        //     console.log(error)
-        //     res.status(500).send();
-        // }
+    signupUser: async (req, res, next) => {
         try {
             const { name, username, password } = req.body;
-            console.log("name", name)
 
             // Check if the username is already taken
             const existingUser = await User.findOne({ where: { username } });
             if (existingUser) {
                 return res.status(400).json({ message: 'Username already in use' });
             }
 
-            // Create the user
-            const newUser = await User.create({ name, username, password });
-            const token = jwt.sign({ id: newUser.id, username: newUser.username }, SECRET_KEY, {
-                expiresIn: '1h', // Token expiration time
+            //create user
+            const hashedPassword = await bcrypt.hash(password, 10);
+            const newUser = await User.create({ name, username, password: hashedPassword });
+
+            // Add default role and permission
+            const role = await Role.findByPk(2); // Assuming role with id 1 is the default role
+            if (role) {
+                await newUser.addRole(role);
+            }
+
+            const permission = await Permission.findByPk(2); // Assuming permission with id 1 is the default permission
```

we shouldnt be doing anything with permission here,
we are assuming that we already have a user role that has the required permissions already assigned

---

when user create acount , it has default permission and role id that show it's User , Just One User has Admin Role, so it's correct

---

we are assuming roles and permissions are predefined in our database. this means there will always be a role called "user" that has certain permissions already assigned to it therefore we need to assign that predefined role to the user when it is being created.

concerning what you mentioned on pumble, you can also add a check to see if the role exists then add it to the user and if not, it should throw an error in the API saying the role does not exist.
for this, you should start studying a concept called `Transaction` in Seqeulize's docs.
what this does, is it reverts any database operation, if any error occurs, so you wouldn't need to worry about creating a user and then throwing an error about the role.

---

we have to dicause about this